tsl/src/bgw_policy/policy_utils.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * This file and its contents are licensed under the Timescale License.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-TIMESCALE for a copy of the license.
+ */
+
+#include <postgres.h>
+#include <utils/builtins.h>
+#include "dimension.h"
+#include "jsonb_utils.h"
+#include "policy_utils.h"
+
+/* Helper function to compare jsonb label value in the config
+ * with passed in value.
+ * This function is used for labels defined on the hypertable's dimension
+ * Parameters:
+ * config - jsonb config value
+ * label - label we are looking for inside the config
+ * partitioning_type - Oid for hypertable's dimension column
+ * lag_value - value we will compare against the config's
+ *             value for the label
+ * lag_type - Oid for lag_value
+ * Returns:
+ *    True, if config value is equal to lag_value
+ */
+bool
+policy_config_check_hypertable_lag_equality(Jsonb *config, const char *json_label,
+											Oid partitioning_type, Oid lag_type, Datum lag_datum)
+{
+	if (IS_INTEGER_TYPE(partitioning_type))
+	{
+		bool found;
+		int64 config_value = ts_jsonb_get_int64_field(config, json_label, &found);
+
+		if (!found)
+			ereport(ERROR,
+					(errcode(ERRCODE_INTERNAL_ERROR),
+					 errmsg("could not find %s in config for existing job", json_label)));
+
+		switch (lag_type)
+		{
+			case INT2OID:
+				return config_value == DatumGetInt16(lag_datum);
+			case INT4OID:
+				return config_value == DatumGetInt32(lag_datum);
+			case INT8OID:
+				return config_value == DatumGetInt64(lag_datum);
+			default:
+				return false;
+		}
+	}
+	else
+	{
+		if (lag_type != INTERVALOID)
+			return false;
+		Interval *config_value = ts_jsonb_get_interval_field(config, json_label);
+		if (config_value == NULL)
+			ereport(ERROR,
+					(errcode(ERRCODE_INTERNAL_ERROR),
+					 errmsg("could not find %s in config for job", json_label)));
+
+		return DatumGetBool(
+			DirectFunctionCall2(interval_eq, IntervalPGetDatum(config_value), lag_datum));
+	}
+}
+
+Datum
+subtract_integer_from_now(int64 interval, Oid time_dim_type, Oid now_func)
+{
+	AssertArg(IS_INTEGER_TYPE(time_dim_type));
+
+	Datum now = OidFunctionCall0(now_func);
+	Datum result = DirectFunctionCall2(int8mi, now, Int64GetDatum(interval));
+	int64 timevalue = DatumGetInt64(result);
+
+	/* Downcast to right type */
+	switch (time_dim_type)
+	{
+		case INT2OID:
+			if (timevalue < PG_INT16_MIN || timevalue > PG_INT16_MAX)
+				ereport(ERROR,
+						(errcode(ERRCODE_NUMERIC_VALUE_OUT_OF_RANGE),
+						 errmsg("subtract integer from now smallint out of range")));
```

Comment:
@erimatnor  This is really an internal error since this happens during policy handling. The user will not be directly interacting with the command. So these error messages are mostly diagnostics for support when policies fail . These type of issues typically end up as bugs for dev. More context is useful in these scenarios.
I have so far never seen user logs with file and line numbers when we debug bgw issues.
If you still strongly think that this should be a generic error message, I'll revert.